src/python/T0/StorageManager/StorageManagerAPI.py

```diff
@@ -87,6 +89,15 @@ def injectNewData(dbInterfaceStorageManager,
 
         if run not in newRunStreams:
             newRunStreams[run] = set()
+        
+        # If is a master agent, and the stream is included in the ignore list, dont inject it
+        # If is a slave agent, and the stream is injected by master (not in masterIgnoreStreams), it should ignore it
+        if isMasterAgent and stream in masterIgnoreStreams:
+            continue
+        
+        if not isMasterAgent and stream not in masterIgnoreStreams:
+            continue
```

Comment:
This is preventing the agent to add the new run/stream combination, but we are still adding the streamer files.

Reply:
We need to remove files that belong to the ignored streams from `newData` before this `for` loop starts.